src/components/task-step/exercise-multiple-choice.cjsx

```diff
@@ -164,41 +176,32 @@ module.exports = React.createClass
     goToStep: React.PropTypes.func.isRequired
     onNextStep: React.PropTypes.func.isRequired
 
-  render: ->
-    {id} = @props
-    step = TaskStepStore.get(id)
-    {id, content, free_response, is_completed} = step
-    # TODO: Assumes 1 question.
-    hasFreeResponse = TaskStepStore.hasFreeResponse(id)
-
-    # This should handle the 4 different states an Exercise can be in:
-    # 1. `not(free_response)`: Show the question stem and a text area
-    # 2. `free_response and not(is_completed)`: Show stem, your free_response, and the multiple choice options
-    # 3. `correct_answer`: review how you did and show feedback (if any)
-    # 4. `task.is_completed and answer` show your answer choice but no option to change it
-
-    # This should also handle when an Exercise format is a True-False:
-    # 5.  `question.formats` does not have 'free-response' and not(is_completed): Show stem and true-false options
-    # 6.  `question.formats` does not have 'free-response' and `correct_answer`: review how you did and show feedback (if any)
-
-    if is_completed
-      # 3. `correct_answer`: review how you did and show feedback (if any)
-      # 6.  `question.formats` does not have 'free-response' and `correct_answer`: review how you did and show feedback (if any)
-      <ExerciseReview
+  renderReview: (id)->
+    <ExerciseReview
         id={id}
         onNextStep={@props.onNextStep}
         goToStep={@props.goToStep}
         onStepCompleted={@props.onStepCompleted}
-      />
-    else if free_response or not hasFreeResponse
-      # 2. `free_response and not(is_completed)`: Show stem, your free_response, and the multiple choice options
-      # 5.  `question.formats` does not have 'free-response' and not(is_completed): Show stem and true-false options
-      <ExerciseMultiChoice
-        id={id}
-        onStepCompleted={@props.onStepCompleted}
-      />
-    else
-      # 1. `not(free_response)`: Show the question stem and a text area
-      <ExerciseFreeResponse
-        id={id}
-      />
+    />
+
+  renderMultipleChoice: (id)->
+    <ExerciseMultiChoice
+      id={id}
+      onStepCompleted={@props.onStepCompleted}
+      onNextStep={@props.onNextStep}
+    />
+
+  renderFreeResponse: (id)->
+    <ExerciseFreeResponse
+      id={id}
+    />
+
+  # add render methods for different panel types as needed here
+
+  render: ->
+    {id} = @props
+    # get panel to render based on step progress
+    panel = StepPanelStore.getPanel id
+    # panel is one of ['review', 'multiple-choice', 'free-response']
+    renderPanelMethod = camelCase "render-#{panel}"
+    @[renderPanelMethod]?(id)
```

Comment:
Is there a reason this should be silently ignored? (Worried about bugs and more types of questions)